src/averageAges.js

```diff
@@ -55,7 +68,31 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const getPeople = onlyWithSon
+    ? people.filter((person) => person.sex === 'm')
+    : people;
```

Comment:
```suggestion
  const getPeople = onlyWithSon
    ? people.filter((person) => person.sex === 'm')
    : people.filter(child =>
      people.find(person => person.name === child.mother));
```
You could easily filter inmeadiatly for your the element you are looking to work with we can call it child and then you find the child by looking that person that has a mother, by comparing each person name, with every element mother . and you could save some coding afterwards.